ls1intum/Artemis
# Feature/exam multi correction/exam multi correction

### Checklist


### Motivation and Context

This PR will be a followup to [https://github.com/ls1intum/Artemis/pull/2594](url). It completes the changes to the assessment dashboard and all the underlying and necessary changes on the server.
It resolves occurring issues with displaying submissions. (Assessed / Unassessed ones) 

### Description

The following changes are done within this PR:

- All behavior related to non-Exam Exercises and views should be exactly as it was
- Progress bars are updated to work with a second correction
- Assessment dashboard display bug-fixes
- query bug-fixes
- server and client tests

### Steps for Testing

1. Create an exam with 1 or 2 correction rounds
2. create Text-, Modeling-, Quiz-, and Programming Exercises
3. assign students
4. participate in the exam
5. assess the exam with:
correctionRounds=1: everything should work as normal
correctionRounds=2: 
   - the second progressbar should be shown, aswell the Correction Round 2 
   - after an assesssment in the first round, the possibility for a new assessment under second correction should appear. This button does not have its functionality yet!
   - During assessment try: Locking, Overriding, Submitting



### Test Coverage

    tutor-participation-graph: 100%
    exercise-assessment-dashboard: 84.15%

### Screenshots
![image](https://user-images.githubusercontent.com/33342534/103483434-7fb0a800-4de7-11eb-9ab7-997c1bd11c6a.png)
